weave/trace/util.py

```diff
@@ -0,0 +1,76 @@
+from concurrent.futures import ThreadPoolExecutor as _ThreadPoolExecutor
+from contextvars import Context, copy_context
+from functools import partial
+from threading import Thread as _Thread
+from typing import Any, Callable, Iterable, Iterator, Optional
+
+
+class ContextAwareThreadPoolExecutor(_ThreadPoolExecutor):
+    """A ThreadPoolExecutor that runs functions with the context of the caller.
```

Comment:
I see you have added some comments, but as a user, I still don't know when/why to use this. You have described _what_ this does from a developer's perspective, but if an external user reads this I don't think they will quite understand what problem it is solving or why to use it